accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,366 @@
+
+### Add and removal of nodes to the cluster
+
+SUSE Manager also offers the possibility of adding and removing nodes to the cluster:
+
+- To add a node from scratch, an AutoYaST profile can be reused
+- To remove a node, SUSE Manager must remove the node from the cluster using the Management Node and issuing the proper procedure to restrict a node off the cluster
+
+SUSE Manager should be instructed to perform additional actions depending on the role of added/removed node (e.g. control plane, worker, load balancer, ...).
+
+### Cluster upgrade
+
+If every node of the cluster is not directly registered to SUSE Manager, SUSE Manager should offer the possibility to use the Management Node to trigger an update of every node in the cluster, following the standard procedure of upgrading predisposed by the cluster provider.
+
+Also, SUSE Manager must coordinate the Management Node to issue a correct upgrade procedure of the product (e.g. from CaaS Platform version 3 to 4).
+
+### Application deployment
+
+SUSE Manager can also offer the possibility of deploying an application on top of the cluster deployed (e.g. Kubernetes deployment descriptor to deploy a Kubernetes operator). In this case, the cluster provider must provide the Management Node of the necessary tooling to deploy an application using an API. SUSE Manager makes use of the Management Node to deploy a provided application.
+
+### Specific usage of SUSE Manager features
+
+One cluster deployment can also leverage SUSE Manager features: SUSE CaaS Platform can trigger a CVE Audit on a container image before deploying it. The cluster provider must outline which features intend to use and which interface is to be used (preferred: XMLRPC API).
+
+## Specific section: manage a CaaS Platform cluster using SUSE Manager
+
+### Show cluster status
+
+SUSE Manager shows the status of the cluster by invoking `skuba cluster status` on the Management Node (via Salt).
+
+E.g. of the output:
+
+```
+NAME         OS-IMAGE                              KERNEL-VERSION        CONTAINER-RUNTIME   HAS-UPDATES   HAS-DISRUPTIVE-UPDATES
+cp0       SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+worker0   SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+```
+
+The output is presented in the "CaaS Platform > cluster name" section.
+
+### Removing nodes from the cluster
+
+A node can also be removed from the cluster:
+
+- temporarily (drained and cordoned off): [using `kubectl`](https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/#use-kubectl-drain-to-remove-a-node-from-service) from the Management Node via Salt.
+- permanently: by invoking `skuba node remove <nodename>` via Salt. NOTE: this removal is permanent and the cluster cannot join any cluster in the future (a complete reinstallation is required to so).
```

Comment:
It would be nice to have SUSE Manager invoke `skuba node remove` and then, if the user wants, reprovision the node from scratch (I know SUMA has this capability).

Reply:
Ok, I added the SUSE Manager redeployment of the node.